ingestion/src/metadata/ingestion/source/objectstore/s3/metadata.py

```diff
@@ -0,0 +1,287 @@
+        return 0
+
+    def _load_metadata_file(self, bucket_name: str) -> Optional[ObjectStoreContainerConfig]:
+        """
+        Load the metadata template file from the root of the bucket, if it exists
+        """
+        if self._is_metadata_file_present(bucket_name=bucket_name):
+            try:
+                logger.info(f"Found metadata template file at - s3://{bucket_name}/{OPENMETADATA_TEMPLATE_FILE_NAME}")
+                response_object = self.s3_client.get_object(Bucket=bucket_name, Key=OPENMETADATA_TEMPLATE_FILE_NAME)
+                content = json.load(response_object['Body'])
+                metadata_config = ObjectStoreContainerConfig.parse_obj(content)
+                return metadata_config
+            except Exception as e:
+                logger.debug(traceback.format_exc())
+                logger.warning(f"Failed loading metadata file s3://{bucket_name}/{OPENMETADATA_TEMPLATE_FILE_NAME}-{e}")
+        return None
+
+    def _is_metadata_file_present(self, bucket_name: str):
+        return prefix_exits(self.s3_client, bucket_name=bucket_name, prefix=OPENMETADATA_TEMPLATE_FILE_NAME)
+
+    def _generate_unstructured_container(self, bucket_response: S3BucketResponse) -> S3ContainerDetails:
+        return S3ContainerDetails(
+            name=bucket_response.Name,
+            CreationDate=bucket_response.CreationDate.isoformat(),
+            numberOfObjects=self._fetch_metric(bucket_name=bucket_response.Name, metric=S3Metric.NUMBER_OF_OBJECTS),
+            size=self._fetch_metric(bucket_name=bucket_response.Name, metric=S3Metric.BUCKET_SIZE_BYTES),
+            fileFormats=[],  # TODO should we fetch some random files by extension here? Would it be valuable info?
+            dataModel=None
+        )
+
+    def _get_sample_file_path(self, bucket_name: str, metadata_entry: MetadataEntry) -> Optional[str]:
+        """
+        Given a bucket and a metadata entry, returns the full path key to a file which can then be used to infer schema
+        or None in the case of a non-structured metadata entry, or if no such keys can be found
+        """
+        result = f"{metadata_entry.dataPath.strip('/')}"
+        if not metadata_entry.isStructured:
+            return None
+        if metadata_entry.isPartitioned and metadata_entry.partitionColumn:
+            result = f"{result}/{metadata_entry.partitionColumn.strip('/')}"
+
+        response = self.s3_client.list_objects_v2(Bucket=bucket_name, Prefix=result)
+        # no objects found in the data path
+        if 'Contents' not in response:
+            return None
+        # this will look only in the first 1000 files under that path (default for list_objects_v2).
+        # We'd rather not do pagination here as it would incur unwanted costs
+        candidate_keys = [
+            entry['Key'] for entry in response['Contents'] if
+            entry and entry['Key'] and entry['Key'].endswith(metadata_entry.structureFormat)
```

Comment:
if `Key` is not a key in `entry` this will blow up. Maybe check `entry.get("Key")`